"<!doctype html>\n<html lang=en>\n  <head>\n    <title>sqlalchemy.orm.exc.UnmappedInstanceError: Class 'pkg.myforms.Comment' is not mapped\n // Werkzeug Debugger</title>\n    <link rel=\"stylesheet\" href=\"?__debugger__=yes&amp;cmd=resource&amp;f=style.css\">\n    <link rel=\"shortcut icon\"\n        href=\"?__debugger__=yes&amp;cmd=resource&amp;f=console.png\">\n    <script src=\"?__debugger__=yes&amp;cmd=resource&amp;f=debugger.js\"></script>\n    <script>\n      var CONSOLE_MODE = false,\n          EVALEX = true,\n          EVALEX_TRUSTED = false,\n          SECRET = \"aWo6o4Q2d6slHFTfGCDx\";\n    </script>\n  </head>\n  <body style=\"background-color: #fff\">\n    <div class=\"debugger\">\n<h1>UnmappedInstanceError</h1>\n<div class=\"detail\">\n  <p class=\"errormsg\">sqlalchemy.orm.exc.UnmappedInstanceError: Class &#39;pkg.myforms.Comment&#39; is not mapped\n</p>\n</div>\n<h2 class=\"traceback\">Traceback <em>(most recent call last)</em></h2>\n<div class=\"traceback\">\n  <h3></h3>\n  <ul><li><div class=\"frame\" id=\"frame-4481757392\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py\"</cite>,\n      line <em class=\"line\">2619</em>,\n      in <code class=\"function\">add</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">        </span>&#34;&#34;&#34;</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>if _warn and self._warn_on_events:</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>self._flush_warning(&#34;Session.add()&#34;)</pre>\n<pre class=\"line before\"><span class=\"ws\"></span> </pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>try:</pre>\n<pre class=\"line current\"><span class=\"ws\">            </span>state = attributes.instance_state(instance)</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>except exc.NO_STATE as err:</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>util.raise_(</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>exc.UnmappedInstanceError(instance),</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>replace_context=err,</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>)</pre></div>\n</div>\n\n<li><div class=\"exc-divider\">The above exception was the direct cause of the following exception:</div>\n<li><div class=\"frame\" id=\"frame-4384941312\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">2548</em>,\n      in <code class=\"function\">__call__</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">    </span>def __call__(self, environ: dict, start_response: t.Callable) -&gt; t.Any:</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>&#34;&#34;&#34;The WSGI server calls the Flask application object as the</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>WSGI application. This calls :meth:`wsgi_app`, which can be</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>wrapped to apply middleware.</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>&#34;&#34;&#34;</pre>\n<pre class=\"line current\"><span class=\"ws\">        </span>return self.wsgi_app(environ, start_response)</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386059456\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">2528</em>,\n      in <code class=\"function\">wsgi_app</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">            </span>try:</pre>\n<pre class=\"line before\"><span class=\"ws\">                </span>ctx.push()</pre>\n<pre class=\"line before\"><span class=\"ws\">                </span>response = self.full_dispatch_request()</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>except Exception as e:</pre>\n<pre class=\"line before\"><span class=\"ws\">                </span>error = e</pre>\n<pre class=\"line current\"><span class=\"ws\">                </span>response = self.handle_exception(e)</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>except:  # noqa: B001</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>error = sys.exc_info()[1]</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>raise</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>return response(environ, start_response)</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>finally:</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386059568\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">2525</em>,\n      in <code class=\"function\">wsgi_app</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">        </span>ctx = self.request_context(environ)</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>error: t.Optional[BaseException] = None</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>try:</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>try:</pre>\n<pre class=\"line before\"><span class=\"ws\">                </span>ctx.push()</pre>\n<pre class=\"line current\"><span class=\"ws\">                </span>response = self.full_dispatch_request()</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>except Exception as e:</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>error = e</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>response = self.handle_exception(e)</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>except:  # noqa: B001</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>error = sys.exc_info()[1]</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386059680\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">1822</em>,\n      in <code class=\"function\">full_dispatch_request</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">            </span>request_started.send(self)</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>rv = self.preprocess_request()</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>if rv is None:</pre>\n<pre class=\"line before\"><span class=\"ws\">                </span>rv = self.dispatch_request()</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>except Exception as e:</pre>\n<pre class=\"line current\"><span class=\"ws\">            </span>rv = self.handle_user_exception(e)</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>return self.finalize_request(rv)</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">    </span>def finalize_request(</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>self,</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>rv: t.Union[ft.ResponseReturnValue, HTTPException],</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386059792\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">1820</em>,\n      in <code class=\"function\">full_dispatch_request</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\"></span> </pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>try:</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>request_started.send(self)</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>rv = self.preprocess_request()</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>if rv is None:</pre>\n<pre class=\"line current\"><span class=\"ws\">                </span>rv = self.dispatch_request()</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>except Exception as e:</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>rv = self.handle_user_exception(e)</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>return self.finalize_request(rv)</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">    </span>def finalize_request(</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386059904\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\"</cite>,\n      line <em class=\"line\">1796</em>,\n      in <code class=\"function\">dispatch_request</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">            </span>and req.method == &#34;OPTIONS&#34;</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>):</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>return self.make_default_options_response()</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span># otherwise dispatch to the handler for that endpoint</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>view_args: t.Dict[str, t.Any] = req.view_args  # type: ignore[assignment]</pre>\n<pre class=\"line current\"><span class=\"ws\">        </span>return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">    </span>def full_dispatch_request(self) -&gt; Response:</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>&#34;&#34;&#34;Dispatches the request and on top of that performs request</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>pre and postprocessing as well as HTTP exception catching and</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>error handling.</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4386060016\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/pkg/myroutes/user_route.py\"</cite>,\n      line <em class=\"line\">164</em>,\n      in <code class=\"function\">theconversations</code></h4>\n  <div class=\"source \"><pre class=\"line before\"><span class=\"ws\">            </span>return render_template(&#39;user/theconversations.html&#39;,all_post=all_post)</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>else:</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>com = request.form.get(&#39;comment&#39;)</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>userid= session.get(&#39;loggedin&#39;)</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>comment = Comment(comment_by= userid, comment_content=com,comment_postid =id)</pre>\n<pre class=\"line current\"><span class=\"ws\">            </span>db.session.add(comment)</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>db.session.commit()</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>data = {&#39;madeby&#39;:comment.commentby.user_fname, &#34;comment&#34;:com}</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>data_json = jsonify(data)</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4385883856\">\n  <h4>File <cite class=\"filename\">\"&lt;string&gt;\"</cite>,\n      line <em class=\"line\">2</em>,\n      in <code class=\"function\">add</code></h4>\n  <div class=\"source \"></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4385883968\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py\"</cite>,\n      line <em class=\"line\">2621</em>,\n      in <code class=\"function\">add</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">            </span>self._flush_warning(&#34;Session.add()&#34;)</pre>\n<pre class=\"line before\"><span class=\"ws\"></span> </pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>try:</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>state = attributes.instance_state(instance)</pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>except exc.NO_STATE as err:</pre>\n<pre class=\"line current\"><span class=\"ws\">            </span>util.raise_(</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>exc.UnmappedInstanceError(instance),</pre>\n<pre class=\"line after\"><span class=\"ws\">                </span>replace_context=err,</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>)</pre>\n<pre class=\"line after\"><span class=\"ws\"></span> </pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>self._save_or_update_state(state)</pre></div>\n</div>\n\n<li><div class=\"frame\" id=\"frame-4481748992\">\n  <h4>File <cite class=\"filename\">\"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/util/compat.py\"</cite>,\n      line <em class=\"line\">208</em>,\n      in <code class=\"function\">raise_</code></h4>\n  <div class=\"source library\"><pre class=\"line before\"><span class=\"ws\">            </span># intermediary exception does not change, but we can&#39;t figure</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span># that out.</pre>\n<pre class=\"line before\"><span class=\"ws\">            </span>exception.__cause__ = replace_context</pre>\n<pre class=\"line before\"><span class=\"ws\"></span> </pre>\n<pre class=\"line before\"><span class=\"ws\">        </span>try:</pre>\n<pre class=\"line current\"><span class=\"ws\">            </span>raise exception</pre>\n<pre class=\"line after\"><span class=\"ws\">        </span>finally:</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span># credit to</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span># https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span># as the __traceback__ object creates a cycle</pre>\n<pre class=\"line after\"><span class=\"ws\">            </span>del exception, replace_context, from_, with_traceback</pre></div>\n</div>\n</ul>\n  <blockquote>sqlalchemy.orm.exc.UnmappedInstanceError: Class &#39;pkg.myforms.Comment&#39; is not mapped\n</blockquote>\n</div>\n\n<div class=\"plain\">\n    <p>\n      This is the Copy/Paste friendly version of the traceback.\n    </p>\n    <textarea cols=\"50\" rows=\"10\" name=\"code\" readonly>Traceback (most recent call last):\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py&#34;, line 2619, in add\n    state = attributes.instance_state(instance)\nAttributeError: &#39;Comment&#39; object has no attribute &#39;_sa_instance_state&#39;\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 2548, in __call__\n    return self.wsgi_app(environ, start_response)\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 2528, in wsgi_app\n    response = self.handle_exception(e)\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 2525, in wsgi_app\n    response = self.full_dispatch_request()\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 1822, in full_dispatch_request\n    rv = self.handle_user_exception(e)\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 1820, in full_dispatch_request\n    rv = self.dispatch_request()\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py&#34;, line 1796, in dispatch_request\n    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/pkg/myroutes/user_route.py&#34;, line 164, in theconversations\n    db.session.add(comment)\n  File &#34;&lt;string&gt;&#34;, line 2, in add\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py&#34;, line 2621, in add\n    util.raise_(\n  File &#34;/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/util/compat.py&#34;, line 208, in raise_\n    raise exception\nsqlalchemy.orm.exc.UnmappedInstanceError: Class &#39;pkg.myforms.Comment&#39; is not mapped\n</textarea>\n</div>\n<div class=\"explanation\">\n  The debugger caught an exception in your WSGI application.  You can now\n  look at the traceback which led to the error.  <span class=\"nojavascript\">\n  If you enable JavaScript you can also use additional features such as code\n  execution (if the evalex feature is enabled), automatic pasting of the\n  exceptions and much more.</span>\n</div>\n      <div class=\"footer\">\n        Brought to you by <strong class=\"arthur\">DON'T PANIC</strong>, your\n        friendly Werkzeug powered traceback interpreter.\n      </div>\n    </div>\n\n    <div class=\"pin-prompt\">\n      <div class=\"inner\">\n        <h3>Console Locked</h3>\n        <p>\n          The console is locked and needs to be unlocked by entering the PIN.\n          You can find the PIN printed out on the standard output of your\n          shell that runs the server.\n        <form>\n          <p>PIN:\n            <input type=text name=pin size=14>\n            <input type=submit name=btn value=\"Confirm Pin\">\n        </form>\n      </div>\n    </div>\n  </body>\n</html>\n\n<!--\n\nTraceback (most recent call last):\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 2619, in add\n    state = attributes.instance_state(instance)\nAttributeError: 'Comment' object has no attribute '_sa_instance_state'\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 2548, in __call__\n    return self.wsgi_app(environ, start_response)\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 2528, in wsgi_app\n    response = self.handle_exception(e)\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 2525, in wsgi_app\n    response = self.full_dispatch_request()\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 1822, in full_dispatch_request\n    rv = self.handle_user_exception(e)\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 1820, in full_dispatch_request\n    rv = self.dispatch_request()\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/flask/app.py\", line 1796, in dispatch_request\n    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)\n  File \"/Users/victoriaabiodun/Desktop/devconf/pkg/myroutes/user_route.py\", line 164, in theconversations\n    db.session.add(comment)\n  File \"<string>\", line 2, in add\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/orm/session.py\", line 2621, in add\n    util.raise_(\n  File \"/Users/victoriaabiodun/Desktop/devconf/vdevconf/lib/python3.10/site-packages/sqlalchemy/util/compat.py\", line 208, in raise_\n    raise exception\nsqlalchemy.orm.exc.UnmappedInstanceError: Class 'pkg.myforms.Comment' is not mapped\n\n\n-->\n"